src/core/createConfig.js

```diff
@@ -57,9 +57,31 @@ const createConfig = config => {
     /**
      * Validates the configuration against the defined schema.
      */
-    validate: () => {
-      // TODO: Validate existing configuration against defined schema.
-      return true;
+    validate: (schemaObj, key) => {
+      if (!schemaObj) {
+        cfg.validate(cfg.schema);
+        return;
+      }
+      const currentKey = key || "";
```

Comment:
Do we need the last part `|| ""`.
Looks like we are not doing anything when `currentKey` is undefined.